extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -220,16 +207,16 @@ function createDicomWebApi(dicomWebConfig, UserAuthenticationService) {
           (hasAccept ? '' : (hasQuery ? '&' : '?') + `accept=${defaultType}`);
         if (BulkDataURI.indexOf('http') === 0) return acceptUri;
         if (BulkDataURI.indexOf('/') === 0) {
-          return wadoRoot + acceptUri;
+          return `${baseUrl}/${wadoURLPrefix}/${acceptUri}`;
```

Comment:
Create a function on the data access object which gets the URL for a given query type, defaulting to the base URL.  That way there is just a single query generator.

Reply:
You could put the getter function on the config object, so that it can be used for doing things like generating the wadors URI's.  Something like
if( !config.getUrl ) {
  config.getUrl = function(typeId, ...additions)
to get the URL.  It should handle query parameters defined in the base URL or in the suffixes as well....   typeId would be 'wado' or 'qido' etc.